src/plugins/intel_gpu/include/intel_gpu/primitives/ctc_greedy_decoder.hpp

```diff
@@ -29,16 +29,33 @@ struct ctc_greedy_decoder : public primitive_base<ctc_greedy_decoder> {
         , ctc_merge_repeated(ctc_merge_repeated)
         , output_tensor(output_tensor) {}
 
-    uint32_t blank_index;
+    /// @brief Constructs ctc_greedy_decoder primitive.
+    /// @param id This primitive id.
+    /// @param input Input primitive id (input, sequence_indicators, blank_index(optional)).
+    /// @param ctc_merge_repeated Flag for merging repeated labels during the CTC calculation
+    ctc_greedy_decoder(const primitive_id& id,
+                       const std::vector<input_info>& inputs,
+                       const uint32_t blank_index,
+                       const bool ctc_merge_repeated,
+                       const padding& output_padding = padding(),
+                       data_types output_data_type = data_types::i32,
+                       const size_t num_outputs = 1)
+        : primitive_base(id, inputs, {output_padding}, {optional_data_type{output_data_type}}, num_outputs)
+        , blank_index(blank_index)
+        , ctc_merge_repeated(ctc_merge_repeated) {}
+
+    uint32_t blank_index = UINT32_MAX;
     bool ctc_merge_repeated = false;
     tensor output_tensor;
     primitive_id second_output;
+    bool use_multiple_outputs = false;
```

Comment:
I think we don't need this any more too, because now we can distinguish them 1) legacy multiple output (allow_new_shape_infer = false) 2) use_multiple_outputs as your intention (allow_new_shape_infer = true)

Reply:
We can use allow_new_shape_infer for CTCGreedyDecoderSeqLen, but it cannot differentiate it from CTCGreedyDecoder which has only one output.

Reply:
Fixed code not to use use_multiple_outputs in the primitive. Thanks for your comments.